tc/core/polyhedral/memory_promotion.h

```diff
@@ -111,9 +111,10 @@ class TensorReferenceGroup {
   TensorReferenceGroup() {}
 
  public:
-  static TensorGroups accessedBySubtree(
-      const detail::ScheduleTree* tree,
-      const Scop& scop);
+  static TensorGroups accessedWithin(
```

Comment:
Is `static` a sign that this function (and more generally the notion of TensorGroup) is ready for hoisting out of mapped_scop? (not in this PR)

Reply:
It never was in mapped scop.  This is a static constructor of an object that wraps `unique_ptr`s.  We always do it this way.